runtime/compiler/aarch64/codegen/ARM64PrivateLinkage.cpp

```diff
@@ -293,14 +299,396 @@ int32_t TR::ARM64PrivateLinkage::buildPrivateLinkageArgs(TR::Node *callNode,
+            numFloatArgs++;
+            totalSize += TR::Compiler->om.sizeofReferenceAddress();
+            break;
+         case TR::Double:
+            if (numFloatArgs >= numFloatArgRegs)
+               numMemArgs++;
+            numFloatArgs++;
+            totalSize += 2*TR::Compiler->om.sizeofReferenceAddress();
+            break;
+         default:
+            TR_ASSERT(false, "Argument type %s is not supported\n", childType.toString());
+         }
+      }
+
+   // From here, down, any new stack allocations will expire / die when the function returns
+   TR::StackMemoryRegion stackMemoryRegion(*trMemory());
+
+   if (numMemArgs > 0)
+      {
+      pushToMemory = new (trStackMemory()) TR::ARM64MemoryArgument[numMemArgs];
+
+      argMemReg = cg()->allocateRegister();
+      }
+
+   numIntegerArgs = 0;
+   numFloatArgs = 0;
+
+   // Helper linkage preserves all argument registers except the return register
+   // TODO: C helper linkage does not, this code needs to make sure argument registers are killed in post dependencies
+   for (int32_t i = from; (rightToLeft && i >= to) || (!rightToLeft && i <= to); i += step)
+      {
+      TR::MemoryReference *mref = NULL;
+      TR::Register *argRegister;
+      TR::InstOpCode::Mnemonic op;
+
+      child = callNode->getChild(i);
+      childType = child->getDataType();
+
+      switch (childType)
+         {
+         case TR::Int8:
+         case TR::Int16:
+         case TR::Int32:
+         case TR::Address: // have to do something for GC maps here
+            if (child->getDataType() == TR::Address)
+               {
+               argRegister = pushAddressArg(child);
+               }
+            else
+               {
+               argRegister = pushIntegerWordArg(child);
+               }
+            if (numIntegerArgs < numIntArgRegs)
+               {
+               if (!cg()->canClobberNodesRegister(child, 0))
+                  {
+                  if (argRegister->containsCollectedReference())
+                     tempReg = cg()->allocateCollectedReferenceRegister();
+                  else
+                     tempReg = cg()->allocateRegister();
+                  generateMovInstruction(cg(), callNode, tempReg, argRegister);
+                  argRegister = tempReg;
+                  }
+               TR::addDependency(dependencies, argRegister, properties.getIntegerArgumentRegister(numIntegerArgs), TR_GPR, cg());
+               }
+            else // numIntegerArgs >= numIntArgRegs
+               {
+               if (childType == TR::Address)
+                  {
+                  op = TR::InstOpCode::strpostx;
+                  }
+               else
+                  {
+                  op = TR::InstOpCode::strpostw;
+                  }
+               mref = getOutgoingArgumentMemRef(argMemReg, argRegister, op, pushToMemory[argIndex++]);
+               }
+            numIntegerArgs++;
+            break;
+         case TR::Int64:
```

Comment:
Why can't the `TR::Int64` case be combined with the above case?  It looks nearly identical.

Reply:
Merged the `TR::Int64` case with other integer cases.